tests/system/examples/dask/test_dask.py

```diff
@@ -110,3 +112,31 @@ def dask_pipe(x=1, y=10):
 
         # remove compiled dask.yaml file
         os.remove("daskpipe.yaml")
+
+    def test_dask_close(self):
+        self._logger.info("Initializing dask cluster")
+        cluster_start_time = datetime.datetime.now()
+
+        # initialize the dask cluster and get its dashboard url
+        client = self.dask_function.client
+        time_took = (datetime.datetime.now() - cluster_start_time).seconds
+        self._logger.info(
+            "Dask cluster initialization completed", took_in_seconds=time_took
+        )
+
+        worker_start_time = datetime.datetime.now()
+        client.wait_for_workers(self.dask_function.spec.replicas)
+        time_took = (datetime.datetime.now() - worker_start_time).seconds
+        self._logger.info("Workers initialization completed", took_in_seconds=time_took)
+
+        self._logger.info("Shutting Down Cluster")
+        self.dask_function.close()
+        time.sleep(20)  # wait for scheduler and workers to go down
```

Comment:
We should be careful with putting sleeps in our tests, if the time is too long, we're making the the tests long unneeded-ly, if the test will suddenly be slow (slow system, network, etc..) the sleep might not be enough and we'll get transient failures
If you won't sleep here what will happen ? the calls afterwards won't raise ?
if not, let's remove the sleep, if so, let's add a TODO to actually wait here for the pods to move to completed (or just do it now)